misc/python/materialize/checks/rename_view.py

```diff
@@ -27,6 +28,19 @@ def initialize(self) -> Testdrive:
         )
 
     def manipulate(self) -> List[Testdrive]:
+        fix_ownership = (
+            """
+                # When upgrading from old version without roles the views are
+                # owned by default_role, thus we have to change the owner
+                # before dropping them:
+                $ postgres-execute connection=postgres://mz_system:materialize@materialized:6877
+                ALTER VIEW rename_view_viewB2 OWNER TO materialize;
+                ALTER VIEW rename_view_viewA2 OWNER TO materialize;
+                """
+            if self.base_version >= MzVersion.parse("0.46.0-dev")
+            else ""
+        )
```

Comment:
Optional: Just wondering if it makes sense to extract common parts to a generic function. 🤔 What do you think?
If this makes sense, could we either automatically identify the tables and views to be adjusted or hand over these objects to a generic `fix_ownership` function / method?

Reply:
Alright 👍